noxfile.py

```diff
@@ -45,122 +43,62 @@
 maxpython = max(supported_python_versions)
 minpython = min(supported_python_versions)
```

Comment:
Note: The `supported_python_versions` can easily get out of date with `pyproject.toml`.  Are there breadcrumbs to ensure this gets updated when `pyproject.toml` gets updated?  Or, is there a way to scrape `pyproject.toml` for this info?

Reply:
I've been thinking the same thing!  I did see an example `noxfile.py` somewhere that scraped `pyproject.toml` to get the supported versions of Python, but haven't had it be a high priority since I've been pretty thorough when dropping & adding versions of Python.  I raised #2954 so we don't forget.